index.js

```diff
@@ -23,22 +24,44 @@ var plugin = module.exports = function (opts) {
   });
   opts.includeUntested = opts.includeUntested === true;
 
-  var instrumenter = new opts.instrumenter(opts);
+  var defaultInstrumenter = new opts.instrumenter(opts);
 
   return through(function (file, enc, cb) {
+    var instrumenter;
+    var fileContents = file.contents.toString();
+    if (file.sourceMap) {
+      var fileOpts = _.defaultsDeep(_.cloneDeep(opts), {
+        codeGenerationOptions: {
+          sourceMap: file.sourceMap.file,
+          sourceMapWithCode: true,
+          sourceContent: fileContents,
+          sourceMapRoot: file.sourceMap.sourceRoot,
+          file: file.path
+        }
+      });
+      instrumenter = new opts.instrumenter(fileOpts);
```

Comment:
Let's just extend the options in the `if` statement. Then there's no need to do extra logic with the instrumenter as it'll just take in the options.